portal-web/test/functional/com/liferay/portalweb/tests/enduser/analyticscloud/GenerateToken.testcase

```diff
@@ -0,0 +1,79 @@
+@component-name = "portal-analytics-cloud"
+definition {
+
+	property analytics.cloud.release = "true";
+	property analytics.cloud.upstream = "true";
+	property custom.properties = "jsonws.web.service.paths.excludes=";
+	property portal.release = "false";
+	property portal.upstream = "true";
+	property property.group = "analytics.cloud.profile";
+	property testray.main.component.name = "Analytics Cloud";
+
+	setUp {
+		task ("Copy the token from AC") {
+			ACUtils.launchAC();
+		}
+
+		task ("Navigate to settings > Api") {
+			ACNavigation.goToSettings();
+
+			ACSettings.goToApis();
+		}
+	}
+
+	tearDown {
+		var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+		ACUtils.tearDownDXP();
+
+		ACUtils.launchAC();
+
+		ACNavigation.goToSettings();
+
+		ACSettings.goToApis();
+
+		ACUtils.clickAnyButton(button = "Revoke");
+
+		ACUtils.clickAnyButton(button = "Continue");
+
+		// It is not possible to delete properties and data source after LRAC-13389. This functionality should return after LRAC-8040
+
+		//ACUtils.tearDownAC();
+
+		if (${testPortalInstance} == "true") {
+			PortalInstances.tearDownCP();
+		}
+		else {
+			JSONUser.tearDownNonAdminUsers();
+
+			JSONGroup.deleteGroupByName(groupName = "Site Name");
+		}
+	}
+
+	@description = "Feature ID: LRAC-13255 | Automation ID: LRAC-13688 | Test Summary: Revoke a token then generate an API token that expires in 30 days"
+	@priority = 3
+	test GenerateTokenWithExpirationDate30Days {
+		task ("Given: An API token with 30 day expiration date is generated") {
+			ACApi.configureExpirationDate(expirationDate = "30 Days");
+
+			ACUtils.clickAnyButton(button = "Generate Token");
+
+			Alert.viewSuccessMessage();
+		}
+
+		task ("When: The token is revoked and a new token is generated") {
+			ACUtils.clickAnyButton(button = "Revoke");
+
+			ACUtils.clickAnyButton(button = "Continue");
+
+			ACApi.configureExpirationDate(expirationDate = "30 Days");
+
+			ACUtils.clickAnyButton(button = "Generate Token");
```

Comment:
Hey Nick, I don't think this part of the code is necessary since the aim of the test is to:
```
1. Generate the token with the expiration date to 30 days
2. Assert the message of the token created
3. Assert the card to create a new token is not visible
``` 
You are already doing most of the test steps you just need to do the step 3 :slightly_smiling_face: 
If I can help you anything let me know